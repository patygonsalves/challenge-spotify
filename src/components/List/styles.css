
.listX-content {
  display: flex; 
  flex-direction: column;
  align-items: center;
  padding: 10px;
}

.listX-title {
  font-size: 14px;
  font-weight: bold;
  color: #fff;
  padding: 5px;
  text-align: center;
}

.listX-description {
  font-size: 12px;
  color: #fff;
}