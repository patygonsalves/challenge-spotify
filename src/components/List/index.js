import React from 'react'
import PropTypes from 'prop-types'

import './styles.css'

const ListItem = ({
  title,
  description,
  image,
  onClick,
}) => (
  <li onClick={onClick}>
    <div className="listX-content">
      {image && <img src={image} alt={`Image by ${title}`} width={145} height={145} />}
      <span className='listX-title'>{title}</span>
      <span className='listX-description'>{description}</span>
    </div>
  </li>
)

ListItem.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  onClick: PropTypes.func,
}

export default ListItem
