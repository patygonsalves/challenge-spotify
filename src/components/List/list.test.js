import React from 'react'
import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'

import List from '.'

describe('<List />', () => {
  test('should render SearchField', () => {
    const { container } = render(
      <List
        title={'Taylor Swift'}
        description={'folklore'}
        image={'http://imagetaylor'}
      />
    )

    expect(screen.getByText(/taylor swift/i)).toBeInTheDocument()
    expect(screen.getByText(/folklore/i)).toBeInTheDocument()
    expect(screen.getByAltText(/Image by taylor swift/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })


})
