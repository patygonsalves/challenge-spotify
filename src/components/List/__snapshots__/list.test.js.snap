// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render SearchField 1`] = `
<li
  class="listX-container"
>
  <div
    class="listX-content"
  >
    <img
      alt="Image by Taylor Swift"
      height="128"
      src="http://imagetaylor"
      width="128"
    />
    <span
      class="listX-title"
    >
      Taylor Swift
    </span>
    <span
      class="listX-description"
    >
      folklore
    </span>
  </div>
</li>
`;
